normalizer_test.py

```diff
@@ -0,0 +1,84 @@
+# Lint as: python3
+"""Tests for evaluating text normalizer."""
```

Comment:
The unit test looks great!  You could try using the parameterized tests that Sandy had previously mentioned!  Here's an example:
https://pypi.org/project/parameterized/

Reply:
For this and the starter project I used `subtest` from the `unittest` module (https://docs.python.org/3/library/unittest.html), which I _think_ does the same thing. For each of the tests (e.g. `test_remove_extra_whitespace`), if one of the test_cases fails, the whole test doesn't break; instead it keeps running the rest of the test_cases and reports the individual cases that fail rather than just saying the whole test failed.

If `parameterized` is what you actually use / what you are more familiar with though, I can definitely try to switch the tests over to that. (Plus it will let me get more experience with other Python modules).

Reply:
Ah cool!  I mean, I don't think anyone's obsessed with the "parameterized" tests, so if this is a good alternative SGTM!  I think the parameterized tests can also end up taking a long time :).